neo4j/internal/bolt/bolt5.go

```diff
@@ -324,14 +325,16 @@ func (b *bolt5) TxBegin(
 	}
 
 	b.queue.appendBegin(tx.toMeta(b.log, b.logId), b.beginResponseHandler())
-	if b.queue.send(ctx); b.err != nil {
-		return 0, b.err
-	}
-	if err := b.queue.receiveAll(ctx); err != nil {
-		return 0, err
-	}
-	if b.err != nil { // onNextMessageErr kicked in
-		return 0, b.err
+	if syncMessages {
+		if b.queue.send(ctx); b.err != nil {
+			return 0, b.err
+		}
+		if err := b.queue.receiveAll(ctx); err != nil {
+			return 0, err
+		}
+		if b.err != nil { // onNextMessageErr kicked in
+			return 0, b.err
+		}
```

Comment:
This check needs to happen regardless if the message is sent or not. `appendBegin` might call the queue's `outgoing.onPackErr` which will set the `b.err`. You might be able to add a unit test to see that in action. Just attach unserializable metadata to the transaction.